### PR TITLE
Create a method to emit event pegout_transaction_created

### PR DESCRIPTION
**Pull request by nathanieliov**
A new method was needed to log the utxo outpoint values using the pegout_transaction_created event. This PR adds this new method for logging/emitting the event.

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/peg/utils/BridgeEventLoggerImpl.java [hunk 1]
```diff
@@ -245,6 +246,21 @@ public void logPegoutConfirmed(Sha256Hash btcTxHash, long pegoutCreationRskBlock
         this.logs.add(new LogInfo(BRIDGE_CONTRACT_ADDRESS, encodedTopics, encodedData));
     }
 
+    @Override
+    public void logPegoutTransactionCreated(Sha256Hash btcTxHash, List<Coin> outpointValues) {
+        if (btcTxHash == null){
+            throw new IllegalArgumentException("btcTxHash param cannot be null");
```
Comment (julia-zack) [inline rskj-core/src/main/java/co/rsk/peg/utils/BridgeEventLoggerImpl.java:252]: `"[logPegoutTransactionCreated] btc transaction hash cannot be null"`

File: rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java [hunk 2]
```diff
@@ -509,6 +515,48 @@ void logPegoutConfirmed() {
...
+    }
+
+    @ParameterizedTest()
+    @MethodSource("logPegoutTransactionCreatedValidArgProvider")
+    void logPegoutTransactionCreated_ok(Sha256Hash btcTxHash, List<Coin> outpointValues) {
+        eventLogger.logPegoutTransactionCreated(btcTxHash, outpointValues);
+        commonAssertLogs(eventLogs);
+
+        assertTopics(2, eventLogs);
+        assertEvent(eventLogs, 0, BridgeEvents.PEGOUT_TRANSACTION_CREATED.getEvent(), new Object[]{btcTxHash.getBytes()}, new Object[]{
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java:537]:
```
eventIndex = 0;
event = BridgeEvents.PEGOUT_TRANSACTION_CREATED.getEvent();
topics = new Object[]{btcTxHash.getBytes()};
data = new Object[]{UtxoUtils.encodeOutpointValues(outpointValues)};
```

use meaningful names

File: rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java [hunk 3]
```diff
@@ -509,6 +515,48 @@ void logPegoutConfirmed() {
...
     }
 
+    private static Stream<Arguments> logPegoutTransactionCreatedValidArgProvider() {
+        List<Arguments> args = new ArrayList<>();
+
+        args.add(Arguments.of(Sha256Hash.ZERO_HASH, Collections.singletonList(Coin.SATOSHI)));
+        args.add(Arguments.of(BitcoinTestUtils.createHash(5), coinListOf(500_000, 400_000, 1_000)));
+        args.add(Arguments.of(BitcoinTestUtils.createHash(10), Collections.singletonList(Coin.ZERO)));
+        args.add(Arguments.of(BitcoinTestUtils.createHash(10), Collections.EMPTY_LIST));
+        args.add(Arguments.of(BitcoinTestUtils.createHash(15), null));
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java:525]:
```
btcTxHash = Sha256Hash.ZERO_HASH;
coinValue = Collections.singletonList(Coin.SATOSHI);
testArguments = Arguments.of(btcTxHash, coinValue);

args.add(testArguments);
```

and so on

File: rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java [hunk 4]
```diff
@@ -509,6 +515,48 @@ void logPegoutConfirmed() {
...
+    }
+
+    private static Stream<Arguments> logPegoutTransactionCreatedInvalidArgProvider() {
+        List<Arguments> args = new ArrayList<>();
+
+        args.add(Arguments.of(null, Collections.singletonList(null), IllegalArgumentException.class));
+        args.add(Arguments.of(null, Collections.singletonList(Coin.SATOSHI), IllegalArgumentException.class));
+        args.add(Arguments.of(Sha256Hash.ZERO_HASH, Collections.singletonList(null), InvalidOutpointValueException.class));
+        args.add(Arguments.of(BitcoinTestUtils.createHash(1), coinListOf(-100), InvalidOutpointValueException.class));
+        args.add(Arguments.of(BitcoinTestUtils.createHash(2), coinListOf(100, -200, 300), InvalidOutpointValueException.class));
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java:549]: same as above

File: rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java (hunk 3 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java:525]: I think what you are proposing is redundant and will make the test cases super hard to follow. It will double the amount of lines that this method currently has. 

Btw, I also consider adding variables to something so obvious like `Collections.singletonList(Coin.SATOSHI);` it is way redundant and unnecessary.

File: rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java (hunk 4 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java:549]: Same as comment as above

File: rskj-core/src/main/java/co/rsk/peg/utils/BridgeEventLoggerImpl.java (hunk 1 above)
Comment (nathanieliov) [reply rskj-core/src/main/java/co/rsk/peg/utils/BridgeEventLoggerImpl.java:252]: That's for logs. This is just an exception.

Comment (julia-zack) [reply rskj-core/src/main/java/co/rsk/peg/utils/BridgeEventLoggerImpl.java:252]: you are right. My bad :)

File: rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java (hunk 2 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java:537]: I agree. Putting these values in variables first will make it easier to read.

**Review by julia-zack — APPROVED** [review]
Approved with one suggestion

File: rskj-core/src/test/java/co/rsk/peg/bitcoin/UtxoTestUtils.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package co.rsk.peg.bitcoin;
+
+import co.rsk.bitcoinj.core.Coin;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public final class UtxoTestUtils {
+    public static List<Coin> coinListOf(long ... values) {
```
Comment (apancorb) [inline rskj-core/src/test/java/co/rsk/peg/bitcoin/UtxoTestUtils.java:9]: Does it make sense to have a class for only this method? Could it be exported from the API of the `UtxoUtilsTest` class so we avoid creating another class?

**Review by apancorb — APPROVED** [review]
LGTM, left minor comment

File: rskj-core/src/test/java/co/rsk/peg/bitcoin/UtxoTestUtils.java (hunk 5 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/bitcoin/UtxoTestUtils.java:9]: You make me think and reconsider it. I guess it would be better to add it to the BitcoinTestUtils class. Thanks!

Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/bitcoin/UtxoTestUtils.java]: Updated!

File: rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java [hunk 6]
```diff
@@ -509,6 +516,51 @@ void logPegoutConfirmed() {
...
+
+        return args.stream();
+    }
+
+    @ParameterizedTest()
+    @MethodSource("logPegoutTransactionCreatedInvalidArgProvider")
+    void logPegoutTransactionCreated_invalidBtcTxHashOrOutpointValues_shouldFail(Sha256Hash btcTxHash, List<Coin> outpointValues, Class<? extends  Exception> expectedException) {
+        assertThrows(expectedException, () -> eventLogger.logPegoutTransactionCreated(btcTxHash, outpointValues));
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java:561]:
```suggestion
    void logPegoutTransactionCreated_invalidBtcTxHashOrOutpointValues_shouldFail(Sha256Hash btcTxHash, List<Coin> outpointValues, Exception expectedException) {
        assertThrows(expectedException.getClass(), () -> eventLogger.logPegoutTransactionCreated(btcTxHash, outpointValues));
```
That looks a bit better

File: rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java [hunk 7]
```diff
@@ -509,6 +516,51 @@ void logPegoutConfirmed() {
...
     }
 
+    private static Stream<Arguments> logPegoutTransactionCreatedValidArgProvider() {
+        List<Arguments> args = new ArrayList<>();
+
+        args.add(Arguments.of(Sha256Hash.ZERO_HASH, Collections.singletonList(Coin.SATOSHI)));
+        args.add(Arguments.of(BitcoinTestUtils.createHash(5), coinListOf(500_000, 400_000, 1_000)));
+        args.add(Arguments.of(BitcoinTestUtils.createHash(10), Collections.singletonList(Coin.ZERO)));
+        args.add(Arguments.of(BitcoinTestUtils.createHash(10), Collections.EMPTY_LIST));
+        args.add(Arguments.of(BitcoinTestUtils.createHash(15), null));
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java:526]: Shouldn't this one throw an exception?

File: rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java (hunk 2 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java]: Done

File: rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java (hunk 7 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java:526]: No, it shouldn't throw any exception due to the current implementation of the [encoding util method](https://github.com/rsksmart/rskj/blob/hsm-segwit-compatibility-integration/rskj-core/src/main/java/co/rsk/peg/bitcoin/UtxoUtils.java#L64). When passing null, it returns an empty array as encoded value for nulls.

Do you think this should be changed to not allow either null or empty list?

File: rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java (hunk 6 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java:561]: This change implies creating an instance of the exception for each entry of the argument provider method. It does not make sense to me to create an instance just to get the class/type of it.

**Review by marcos-iov — APPROVED** [review]
